zerver/lib/actions.py

```diff
@@ -2989,3 +2997,38 @@ def make_dict(row):
     streams.sort(key=lambda elt: elt["name"])
 
     return streams
+
+def do_process_attachment(message):
+
+    def extract_path_id(upload_string):
+        #We need to check if the string within the paranthesis contains a
+        #valid upload string
+        if re.search('[/\-]user[\-_]uploads[/\.-]', upload_string):
+            cleaned_path = re.sub('http://.*?/user_uploads/','', upload_string)
+            cleaned_path = re.sub('[\(\)]', '', cleaned_path)
+            return cleaned_path
+        else:
+            return None
+
+    #get all pairs of paranthesis
+    p = re.compile('\(.*?\)')
```

Comment:
That regex is supposed to stop looking for ')' after seeing the first one, like in a non greedy fashion. That was the behavior I wanted. I should have been more specific on that part. I think I will take a look at that bugdown functions; the one which I wrote might still have some subtle bugs.